### PR TITLE
Use CMake for installation

### PR DESCRIPTION
**Pull request by jtojnar**
- Move install logic to CMake
- Use CMake’s install manifest for uninstallation detection
- Substitute paths in rime.xml

Closes: https://github.com/rime/ibus-rime/issues/108

cc @pmeiyu

### CODE REVIEW

File: rime_config.h [hunk 1]
```diff
@@ -1,10 +0,0 @@
-#ifndef RIME_CONFIG_H_
-#define RIME_CONFIG_H_
-
-#define RIME_VERSION "1.4.0"
-
-#define IBUS_RIME_INSTALL_PREFIX "/usr"
```
Comment (jtojnar) [inline rime_config.h:6]: Does not seem to be used anywhere. Should we still preserve this?

File: rime_config.h.in [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#ifndef RIME_CONFIG_H_
+#define RIME_CONFIG_H_
+
+#define RIME_VERSION "1.4.0"
+
+#define IBUS_RIME_SHARED_DATA_DIR "@RIME_DATA_DIR@"
```
Comment (jtojnar) [inline rime_config.h.in:6]: Not sure if this should point to datadir of RimeData package or rime-ibus package.

They do not have to be installed under the same prefix, for example, on NixOS they will end up in:

* `/nix/store/4hq26wif0my89r1fhm7nlq2c65fniws4-brise-unstable-2017-09-16/share/rime-data/`
* `/nix/store/sgq7jraihcyvw82s8a2x6hpdr8xp2cwk-ibus-rime-1.4.0/share/rime-data/`

File: Makefile [hunk 3]
```diff
@@ -1,8 +1,8 @@
 ifeq (${PREFIX},)
 	PREFIX=/usr
 endif
-sharedir = $(DESTDIR)$(PREFIX)/share
-libexecdir = $(DESTDIR)$(PREFIX)/lib
```
Comment (jtojnar) [inline Makefile:5]: CMake handles `DESTDIR`.

File: Makefile [hunk 4]
```diff
@@ -12,30 +12,19 @@ all: ibus-engine-rime
...
-	install -m 755 -d $(sharedir)/rime-data
-	install -m 644 -t $(sharedir)/rime-data/ ibus_rime.yaml
+	(cd $(builddir); make install)
 
 uninstall:
-	rm $(sharedir)/ibus/component/rime.xml
-	rm -R $(sharedir)/ibus-rime
-	rm -R $(libexecdir)/ibus-rime
-	rm $(sharedir)/rime-data/ibus_rime.yaml
+	(cd $(builddir); xargs rm < install_manifest.txt)
```
Comment (jtojnar) [inline Makefile:27]: This does not remove the directories, only the files. Is it enough? People are probably better off using package manager anyway.

**Review by lotem — APPROVED** [review]
LGTM

File: rime_config.h.in (hunk 2 above)
Comment (jtojnar) [reply rime_config.h.in:6]: Note that I did not test this pull request other than by building as I am not familiar with the input method. This should be checked by someone if the data is still correctly loaded after this change.

**Pull request merged**